File: src/protocol/UserOutOfHomeBuilder.cpp
#include "UserOutOfHomeBuilder.h"
#include "UserOutOfHomeCommand.h"

using namespace std;
using namespace communication;

UserOutOfHomeBuilder::UserOutOfHomeBuilder()
{}

UserOutOfHomeBuilder::~UserOutOfHomeBuilder()
{}

std::unique_ptr<Command> UserOutOfHomeBuilder::create(const std::vector<uint8_t> &commandInBytes)
{
    auto command = make_unique<UserOutOfHomeCommand>();

    return move(command);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/UserOutOfHomeBuilder.java
package wpam.mobile_client.protocol;

import java.io.Closeable;
import java.util.ArrayList;

public final class UserOutOfHomeBuilder implements CommandBuilder, Closeable
	{
		public UserOutOfHomeBuilder()
		{
		}
		public final void close()
		{
		}

		@Override
		public Command create(ArrayList<Integer> commandInBytes)
		{
			Command command  = new UserOutOfHomeCommand();
			return command;
		}
	}